language_service/src/code_lens.rs

```diff
@@ -98,3 +101,16 @@ pub(crate) fn get_code_lenses(
             accum
         })
 }
+
+fn has_blocking_errors(compilation: &Compilation) -> bool {
+    // Don't show code lenses if there are any project errors or compilation errors that prevent execution
+    !compilation.project_errors.is_empty()
+        || compilation.compile_errors.iter().any(|e| {
+            !matches!(
+                e.error(),
+                // These error types don't prevent execution
+                qsc::compile::ErrorKind::Lint(_)
+                    | qsc::compile::ErrorKind::Pass(qsc_passes::Error::EntryPoint(_))
```

Comment:
This check is extraneous; as mentioned in the other thread, the compilation cannot include entry point errors. Once this part of the check is removed the extra dependency also goes away and the changes to the Cargo.toml/lock go away too.
```suggestion
                qsc::compile::ErrorKind::Lint(_))
```